src/circuit/tests.rs

```diff
@@ -1507,6 +1508,119 @@ mod rangecheck {
     }
 }
 
+#[cfg(test)]
+mod rangecheckpercent {
+
+    use crate::tensor::Tensor;
+    use halo2_proofs::{
+        circuit::{Layouter, SimpleFloorPlanner, Value},
+        dev::MockProver,
+        plonk::{Circuit, ConstraintSystem, Error},
+    };
+    use halo2curves::pasta::Fp;
+    use crate::circuit::Tolerance;
+
+    const RANGE: f32 = 0.1; // 0.1 percent error tolerance
+    const K: usize = 8;
+    const LEN: usize = 4;
+
+    use super::*;
+
+    #[derive(Clone)]
+    struct MyCircuit<F: PrimeField + TensorType + PartialOrd> {
+        input: ValTensor<F>,
+        output: ValTensor<F>,
+    }
+
+    impl<F: PrimeField + TensorType + PartialOrd> Circuit<F> for MyCircuit<F> {
+        type Config = BaseConfig<F>;
+        type FloorPlanner = SimpleFloorPlanner;
+        type Params = TestParams;
+
+        fn without_witnesses(&self) -> Self {
+            self.clone()
+        }
+
+        fn configure(cs: &mut ConstraintSystem<F>) -> Self::Config {
+            let a = VarTensor::new_advice(cs, K, LEN);
+            let b = VarTensor::new_advice(cs, K, LEN);
+            let output = VarTensor::new_advice(cs, K, LEN);
```

Comment:
Where does the `GreaterThan` lookup table get configured here ? / do these tests succeed locally from what you can tell ? You can run `cargo test circuit::tests` to run this block of tests